tests/unit/python/foglamp/services/core/api/test_filters.py

```diff
@@ -660,3 +661,246 @@ def q_result(*args):
+        def get_cat(category_name):
+            category = category_name
+            if category == "random1_scale1":
+                return mock_cat
+            if category == 'random1_meta2':
+                return None
+            if category == 'meta2':
+                return mock_cat
+
+        @asyncio.coroutine
+        def create_cat():
+            return {}
+
+        @asyncio.coroutine
+        def create_child_cat():
+            return {}
+
+        storage_client_mock = MagicMock(StorageClientAsync)
+        c_mgr_mock = MagicMock(ConfigurationManager)
+
+        connect_mock = mocker.patch.object(connect, 'get_storage_async', return_value=storage_client_mock)
+        get_category_mock = mocker.patch.object(c_mgr_mock, 'get_category_all_items', side_effect=get_cat)
+        create_category_mock = mocker.patch.object(c_mgr_mock, 'create_category', return_value=create_cat())
+        create_child_category_mock = mocker.patch.object(c_mgr_mock, 'create_child_category',
+                                                         return_value=create_child_cat())
+        update_config_bulk_mock = mocker.patch.object(c_mgr_mock, 'update_configuration_item_bulk',
+                                                      return_value=asyncio.sleep(.1))
+        cache_manager = mocker.patch.object(c_mgr_mock, '_cacheManager')
+        cache_remove = mocker.patch.object(cache_manager, 'remove', return_value=MagicMock())
+        insert_tbl_patch = mocker.patch.object(storage_client_mock, 'insert_into_tbl', return_value=asyncio.sleep(.1))
+
+        # WHEN
+        await filters._add_child_filters(storage_client_mock, c_mgr_mock, user_name_mock, new_list_mock, old_list_mock)
+
+        # THEN
+        calls_get_cat = [call(category_name='random1_scale1'),
+                         call(category_name='random1_meta2'),
+                         call(category_name='meta2')]
+        get_category_mock.assert_has_calls(calls_get_cat, any_order=True)
+
+        calls_create_cat = [
+            call(category_description='Configuration of meta2 filter for user random1', category_name='random1_meta2',
+                 category_value={
+                     'assetName': {'description': 'Name of Asset', 'type': 'string', 'order': '1', 'default': 'test',
+                                   'displayName': 'Asset name'}}, keep_original_items=True)]
+        create_category_mock.assert_has_calls(calls_create_cat, any_order=True)
+
+        calls_create_child = [call(category_name='random1', children=['random1_scale1', 'random1_meta2'])]
+        create_child_category_mock.assert_has_calls(calls_create_child, any_order=True)
+
+        calls_update = [call('random1_meta2', {'assetName': 'test1'})]
+        update_config_bulk_mock.assert_has_calls(calls_update, any_order=True)
+
+        calls_insert = [call('filter_users', '{"user": "random1", "name": "meta2"}')]
+        insert_tbl_patch.assert_has_calls(calls_insert, any_order=True)
```

Comment:
This is failing on mutiple time run at my local and also on CI as well on first commit, please fix the assertion